File: Java/src/test/java/com/gildedrose/GildedRoseTest.java
package com.gildedrose;


import org.assertj.core.api.Assertions;
import org.junit.Test;

public class GildedRoseTest {

    @Test
    public void foo() {
        // Given
        Item[] items = new Item[] { new Item("foo", 0, 0) };
        GildedRose app = new GildedRose(items);

        // When
        app.updateQuality();

        // Then
        Assertions.assertThat(app.items[0].name).isEqualTo("fixme");
    }

}

File: js/spec/gilded_rose.test.js
'use strict';

const {Item, Shop} = require('../src/gilded_rose');

describe("Gilded Rose", function() {

  it("should foo", function() {
    // Given
    const gildedRose = new Shop([ new Item("foo", 0, 0) ]);

    // When
    const items = gildedRose.updateQuality();

    // Then
    expect(items[0].name).toEqual("fixme");
  });

});
